sql: fix wrong representation of SHOW JOBS

spaskob opened the pull request:
`WHEN COMPLETE` was wrongly being put in the end of the string
representation of `SHOW JOBS WHEN COMPLETE SELECT...` stmt.

Fixes: #40534.

Release note (bug fix): None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/40601)
<!-- Reviewable:end -->

Commit:
sql: fix wrong representation of SHOW JOBS

`WHEN COMPLETE` was wrongly being put in the end of the string
representation of `SHOW JOBS WHEN COMPLETE SELECT...` stmt.

Fixes: #40534.

Release note: None

mjibson commented:
The Fixes tag is incorrect in the commit message. It should be `Fixes #XXX`. Missing the # and I think the number is wrong too?

spaskob commented:
Thanks Matt, fixed the incorrect commit message.

spaskob commented:
bors r+
